### ISSUE TITLE
Is there some way of getting a fields tag from its name?

### ISSUE THREAD

**Issue by timthelion** [danburkert/prost#137]
I'd like to be able to write something like

````
get_tag!(message_name, field_name)
````

Rather than writing the numbers directly into my code in places.

**Comment by danburkert** [comment]
No, currently the mapping of fields to tag numbers isn’t exposed.   Can you expand on why you would want to do this?  I’ve never had a need to work with tags directly.

**Comment by timthelion** [comment]
@danburkert It shouldn't be necessary to modify the `proto` file with any sort of annotation. All that is needed is to find out the field's tag number so that the field can be referred to over the wire. This is a simple operation in other protobuf libraries:

-  C++ https://stackoverflow.com/questions/35479900/getting-tag-number-of-a-field-in-google-protobuf

- In python `MessageName.DESCRIPTOR.fields` gives a list of objects with info like the fields name and tag.

- In Golang, protobuf tags are represented as struct tags which can be accessed via reflection: https://medium.com/golangspec/tags-in-golang-3e5db0b8ef3e

I expected that something similar to golang's reflection would exist in rust for just this kind of problem, but it seems that is not the case: https://stackoverflow.com/questions/54573185/how-do-i-read-attributes-programatically-in-rust/54578601?noredirect=1#comment95957359_54578601

Would you accept a PR which appended to the generated code output a simple traversable data-structure which would include this metadata?

**Comment by timthelion** [comment]
Re-reading your comment it appears to me that you misunderstood what I meant by "Configured with protobuf". What I meant, is that a list or map of disallowed field tags is set over the wire to the broker in protobuf format. Not that the broker is configured in a `.proto` file.
